virtualmachine/vsphere/util.go

```diff
@@ -616,7 +620,10 @@ var cloneFromTemplate = func(vm *VM, dcMo *mo.Datacenter, usableDatastores []str
 
 	customizationSpecManager := object.NewCustomizationSpecManager(
 		vm.client.Client)
-	// check if customization specification exists
+
+	checkCustomSpecMutex.Lock()
+	// Critical section - Only one thread should create custom spec
+	// if not present
 	exists, err := customizationSpecManager.DoesCustomizationSpecExist(
 		vm.ctx, STATICIP_CUSTOM_SPEC_NAME)
 	if err != nil {
```

Comment:
should we release the lock in the error case ?

Reply:
can we make it like finally..like defer file close operation so thatb  in case it will execute every time even if error comes or vice-versa.